lib/pbench/server/api/resources/api_key.py

```diff
@@ -92,7 +96,6 @@ def _post(
         POST /api/v1/key?name=name
```

Comment:
This seems minor; but the `POST` doesn't define a `{key}` parameter. The validator deliberately ignores undefined parameters in the JSON body (a deliberate decision I sometimes struggle to justify to myself) and fails with undefined query parameters. But validation of the URI parameters is completely manual. You'd get a 404 if the Flask endpoint didn't define a parameter, because Flask/Werkzeug couldn't map it. However we use a multiplex mapping for this API ... so it's perfectly legal and we're silently ignoring it. I'm not sure I really care much that `POST /key/5` will just create a new key and ignore the ID, but it wouldn't be hard to return an `APIAbort(HTTPStatus.BAD_REQUEST, "you can't choose the key; the key chooses you")` if they URI parameter value isn't missing/empty ...

Reply:
Based on what Dave said, @siddardh-ra, I would encourage you to add the error check.  It's simple-enough to code, and it might help prevent a surprise for some user.  (But, if you decline, I'm not going to block the merge on this point!)